modin/core/execution/dask/implementations/pandas_on_dask/partitioning/virtual_partition.py

```diff
@@ -372,6 +372,20 @@ def length(self):
         """
         if self._length_cache is None:
             if self.axis == 0:
+                caches = [
+                    obj.try_build_length_cache()
+                    for obj in self.list_of_partitions_to_combine
+                ]
+                new_lengths = DaskWrapper.materialize(
+                    [cache for cache in caches if isinstance(promise, Future)]
+                )
+                dask_idx = 0
+                for i, cache in enumerate(caches):
+                    if isinstance(cache, Future):
+                        self.list_of_partitions_to_combine[i].try_set_length_cache(
+                            new_lengths[dask_idx]
```

Comment:
Shouldn't this just be `i` as well?

Reply:
No, since `new_lengths` may have fewer elements than `caches` in the case where some length values were already computed (and are filtered out by the `isinstance(cache, Future)` check). The value computed at `new_lengths[dask_idx]` should correspond to the promise at `caches[i]`.